src/coll/mod.rs

```diff
@@ -265,7 +265,11 @@ where
         resolve_options!(self, options, [read_concern, selection_criteria]);
 
         let op = Count::new(self.namespace(), options);
-        self.client().execute_operation(op, None).await
+
+        match self.client().execute_operation(op, None).await {
```

Comment:
The section of the CRUD spec on this functionality indicated that we should return `0` for the count if this method is invoked on a non-existent collection.

Reply:
We can actually use `Operation::handle_error` to do this at the operation layer. We currently do something [similar for `drop_collection`](https://github.com/mongodb/mongo-rust-driver/blob/master/src/operation/drop_collection/mod.rs#L58).

Reply:
Oh, right! I had forgotten about that